src/routes/documents/[id]-[slug]/+page.svelte

```diff
@@ -69,25 +56,6 @@
       }, POLL_INTERVAL);
     }
   });
-
-  $: if (browser) {
-    const u = new URL($page.url);
-    // When the mode changes, update the URL param to match.
-    u.searchParams.set("mode", $currentMode);
-    // Reset the hash, keeping page number if we should paginate.
-    if (shouldPaginate($currentMode) && $currentPage > 1) {
-      u.hash = pageHashUrl($currentPage);
-    } else {
-      u.hash = "";
-    }
-    // Only navigate when the mode changes.
-    // Since we're accessing $page or $currentPage, this will run when
-    // those values change, as well. We want to avoid unnecessary runs.
-    if (previousMode !== $currentMode) {
-      goto(u);
-      previousMode = $currentMode;
-    }
-  }
```

Comment:
I think we can set the `noScroll` option in `goto` to handle this. Will push up that change.

Reply:
When switching between "File" and "Grid" or "Notes" tabs, the current page is lost. I believe this is because the tab `goto`s don't contain any page URL information, and when we update the navigation the current page state is overwritten with the default value.

Reply:
I think that's always been the case. Grid and notes views don't paginate the same way as the document or text.